pkg/threescale/backend/helpers_test.go

```diff
@@ -293,3 +293,144 @@ func Test_ValidateTransactions(t *testing.T) {
 		})
 	}
 }
+
+func TestSynchronizeState(t *testing.T) {
+	original := LimitCounter{
+		"hits": []api.UsageReport{
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Day,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Week,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Month,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Year,
+				},
+			},
+		},
+	}
+
+	new := LimitCounter{
+		"hits": []api.UsageReport{
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Minute,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Day,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Month,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Year,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Eternity,
+				},
+			},
+		},
+	}
+
+	expect := LimitCounter{
+		"hits": []api.UsageReport{
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Minute,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Day,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Month,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Year,
+				},
+			},
+			{
+				PeriodWindow: api.PeriodWindow{
+					Period: api.Eternity,
+				},
+			},
+		},
+	}
+
+	got, _ := synchronizeStates(original, new)
```

Comment:
We would need to test that the changes struct returned here is also correct. Wdyt?